src/jsrs_parser.cc

```diff
@@ -144,65 +148,76 @@ static bool GetType(const char* begin, const char* end, Type* type) {
 
 namespace internal {
 
-const char* PrepareString(Isolate*    isolate,
-                          const char* str,
-                          size_t      length,
-                          size_t*     new_length) {
-  char* result = new char[length + 1];
-  bool string_mode = false;
-  enum { kDisabled = 0, kOneline, kMultiline } comment_mode = kDisabled;
-  size_t j = 0;
-  size_t size = 0;
-
-  for (size_t i = 0; i < length; i++) {
-    if ((comment_mode == kDisabled) &&
-        (str[i] == '\"' || str[i] == '\'') &&
-        (i == 0 || str[i - 1] != '\\')) {
-      string_mode = !string_mode;
+// Returns true if `str` points to a multiline comment ending, false otherwise.
+bool IsMultilineCommentEnd(const char* str, size_t* size) {
+  if (str[0] == '*' && str[1] == '/') {
+    *size = 2;
+    return true;
+  }
+  return false;
+}
+
+// Returns count of bytes needed to skip to current comment ending.
+size_t SkipToCommentEnd(const char* str, const char* end) {
+  bool is_single_line;
+
+  switch (str[1]) {
+    case '/': {
```

Comment:
Well, the Google C++ Styleguide states:

> case blocks in switch statements can have curly braces or not, depending on your preference.

Then we need to decide whether we use them or don't.